tests/src/test/java/io/openshift/booster/OpenshiftIT.java

```diff
@@ -0,0 +1,168 @@
+    private static final String GREETING_SERVICE_APP = "wfswarm-circuit-breaker-greeting";
+
+    private static final String NAME_OK = "{\"state\":\"ok\"}";
+    private static final String NAME_FAIL = "{\"state\":\"fail\"}";
+    private static final String CB_CLOSED = "\"state\":\"closed\"";
+    private static final String CB_OPEN = "\"state\":\"open\"";
+    private static final String HELLO_OK = "Hello, World!";
+    private static final String HELLO_FALLBACK = "Hello, Fallback!";
+
+    // See also circuitBreaker.sleepWindowInMilliseconds
+    private static final long SLEEP_WINDOW = 5000l;
+    // See also circuitBreaker.requestVolumeThreshold
+    private static final long REQUEST_THRESHOLD = 3;
+
+    private static final OpenShiftTestAssistant OPENSHIFT = new OpenShiftTestAssistant();
+
+    private static String nameBaseUri;
+
+    private static String greetingBaseUri;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+
+        nameBaseUri = deployApp(NAME_SERVICE_APP, System.getProperty("nameServiceTemplate"));
+        greetingBaseUri = deployApp(GREETING_SERVICE_APP, System.getProperty("greetingServiceTemplate"));
+
+        await().atMost(5, TimeUnit.MINUTES).until(() -> {
+            List<Pod> list = OPENSHIFT.client().pods().inNamespace(OPENSHIFT.project()).list().getItems();
+            return list.stream()
+                    .filter(pod -> pod.getMetadata().getName().startsWith(NAME_SERVICE_APP) || pod.getMetadata().getName().startsWith(GREETING_SERVICE_APP))
+                    .filter(pod -> "running".equalsIgnoreCase(pod.getStatus().getPhase())).collect(Collectors.toList()).size() >= 2;
+        });
+
+        System.out.println("Pods running, waiting for readiness probe...");
+        String pingUri = greetingBaseUri + "/api/ping";
+
+        await().atMost(5, TimeUnit.MINUTES).until(() -> {
+            try {
+                Response response = get(pingUri);
+                return response.getStatusCode() == 200;
+            } catch (Exception e) {
+                return false;
+            }
+        });
+
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        OPENSHIFT.cleanup();
+    }
+
+    @Test
+    public void testCircuitBreaker() throws InterruptedException {
+        assertCircuitBreaker(CB_CLOSED);
+        assertGreeting(HELLO_OK);
+        changeNameServiceState(NAME_FAIL);
+        for (int i = 0; i < REQUEST_THRESHOLD; i++) {
+            assertGreeting(HELLO_FALLBACK);
+        }
+        // Wait a little to get the current health counts - see also metrics.healthSnapshot.intervalInMilliseconds
+        Thread.sleep(100l);
+        // Circuit breaker should be open now
+        assertCircuitBreaker(CB_OPEN);
+        changeNameServiceState(NAME_OK);
+        System.out.println("Sleep for " + SLEEP_WINDOW + " ms (circuitBreaker.sleepWindowInMilliseconds)");
+        Thread.sleep(SLEEP_WINDOW);
```

Comment:
It wouldn't help much in this case - we have to wait at least for the given interval, i.e. the state cannot be changed sooner.